bench/fixture.ts

```diff
@@ -0,0 +1,325 @@
+            partiallyFillable: true,
+            sellAmount: ethers.utils.parseEther("201.0"),
+            buyAmount: ethers.utils.parseEther("200.0"),
+          };
+          break;
+        default:
+          throw new Error("unreacheable");
+      }
+      const signingScheme =
+        (i + Math.floor(i / 4)) % 2 == 0
+          ? SigningScheme.TYPED_DATA
+          : SigningScheme.MESSAGE;
+      const feeDiscount = (i % 3) * (FULL_FEE_DISCOUNT / 2); // 0% | 50% | 100%
+
+      const dbg = {
+        fill: orderSpice.partiallyFillable
+          ? "partially fillable"
+          : "fill-or-kill",
+        kind: orderSpice.kind == OrderKind.SELL ? "sell" : "buy",
+        sign:
+          signingScheme == SigningScheme.TYPED_DATA ? "typed-data" : "message",
+        fee: 100 * (1 - feeDiscount / FULL_FEE_DISCOUNT),
+      };
+      debug(
+        `encoding ${dbg.fill} ${dbg.kind} order with ${dbg.sign} signature and ${dbg.fee}% fees`,
+      );
+
+      await encoder.signEncodeTrade(
+        {
+          sellToken: tokens.id(i % options.tokens).address,
+          buyToken: tokens.id(i % options.tokens).address,
+          feeAmount: ethers.utils.parseEther("1"),
+          validTo: 0xffffffff,
+          appData: this.nonce++,
+          ...orderSpice,
+        },
+        traders[i % traders.length],
+        signingScheme,
+        {
+          executedAmount: ethers.utils.parseEther("100.0"),
+          feeDiscount,
+        },
+      );
+    }
+
+    for (let i = 0; i < options.interactions; i++) {
+      const [tokenIn, tokenOut] = [i % 2, (i + 1) % 2];
+      encoder.encodeInteraction({
+        target: uniswapTokens[tokenIn].address,
+        callData: uniswapTokens[
+          tokenIn
+        ].interface.encodeFunctionData("transfer", [
+          uniswapPair.address,
+          ethers.utils.parseEther("1.0"),
+        ]),
+      });
+
+      debug(`encoding Uniswap T00${tokenIn} -> T00${tokenOut} token swap`);
+
+      // NOTE: Uniswap pool is large enough that price will pretty much not
+      // move over the course of these benchmarks.
+      const amountOut = ethers.utils.parseEther("0.9");
+      const [amount0Out, amount1Out] =
+        tokenIn == 0 ? [0, amountOut] : [amountOut, 0];
+      encoder.encodeInteraction({
+        target: uniswapPair.address,
+        callData: uniswapPair.interface.encodeFunctionData("swap", [
+          amount0Out,
+          amount1Out,
+          settlement.address,
+          "0x",
+        ]),
+      });
+    }
+
+    if (options.refunds > 0) {
+      debug(`encoding ${options.refunds} order refunds`);
+    }
+    for (let i = 0; i < options.refunds; i++) {
+      const trader = traders[i % traders.length];
+      const orderUid = computeOrderUid({
+        orderDigest: `0x${"42".repeat(32)}`,
+        owner: trader.address,
+        validTo: 0,
+      });
+      await settlement.connect(trader).invalidateOrder(orderUid);
+      encoder.encodeOrderRefunds(orderUid);
+    }
```

Comment:
It looks like if `options.refunds > traders.length` then the orders that are invalidated are the same as before, so the last refunds are setting a zero slot to zero, not providing any refund. You could use `i` in `orderDigest` to have always different orders.

Reply:
:facepalm: Great catch!